### DISCUSSION TITLE
Not an issue, just need help to get this thing working.

### DISCUSSION THREAD

**Discussion by LazyNinja74** [jhuckaby/Cronicle#676] [General]
How to setup a job which will run daily at 1pm. This job should execute a python file which is in a private github repo. Also, is there a terminal kind of thing which would help me to see the output of that python file?

**Comment by jhuckaby** [comment]
This is not an issue, so I'm moving it to a discussion.

**Comment by jhuckaby** [comment]
You can use the built-in [Shell Plugin](https://github.com/jhuckaby/Cronicle/blob/master/docs/Plugins.md#built-in-shell-plugin) to execute arbitrary shell commands, and Cronicle does capture and log the output for you.

You will have to download the python file from your private GitHub repo yourself tho.  This is out of the scope of what Cronicle can do itself.

ChatGPT tells me you can do this in a shell script:

```sh
curl -H "Authorization: token $TOKEN" \
     -H 'Accept: application/vnd.github.v3.raw' \
     -o my-file.py \
     -L https://api.github.com/repos/my-username/my-repo/contents/my-file-path.ext?ref=my-branch

python my-file.py
rm my-file.py
```

Remember to replace `$TOKEN`, `my-file.py`, `my-company`, `my-org`, `my-repo`, `my-file-path.ext`, `my-username`, and `my-branch` with your actual token, desired file name, GitHub domain, organization name, repository name, file path, username, and branch name respectively.
